src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -434,7 +434,9 @@ def parse_error_handler(parse_error):
           for test_name, test_info in tests_info.items():
             test_item = Test(test_info['classname'], test_name)
             test_target = test_registry.get_owning_target(test_item)
-            self.report_test_info(self.options_scope, test_target, test_name, test_info)
+            for test_info_key, test_info_val in test_info.items():
```

Comment:
Could this logic go into report_test_info() (or even better, a function that wraps it), instead of being repeated here and in pytest_run.py?